scripts/cleanup_old_vms.py

```diff
@@ -132,8 +132,19 @@ def scan_provider(provider_key, matchers, match_queue, scan_failure_queue):
         scan_failure_queue.put(VmReport(provider_key, FAIL, NULL, NULL, NULL))
         logger.exception('%r: Exception listing vms', provider_key)
         return
+    text_matched_vms = {}
+    text_matched_duplicate_vms = []
+    for vm in vm_list:
+        if match(matchers, vm.name):
+            if vm.name not in text_matched_vms:
+                text_matched_vms[vm.name] = vm
+            else:
+                text_matched_duplicate_vms.append(vm)
+    text_matched_vms = [vm for vm in text_matched_vms.values()]
+    for vm in text_matched_duplicate_vms:
+        logger.info("%r: DELETING DUPLICATE VMS: %r", provider_key, vm.name)
+        vm.cleanup()
```

Comment:
So this will add a serial vm cleanup call into the threaded call chain that is scanning each provider, greatly slowing the speed of the script, as it won't report a list of VM's to cleanup without first trying to delete any duplicates.

I think we should take a step back from the change you've made here, which is at its core a good approach to the problem at hand.  This was written before wrapanapi3.0, and is passing around a VM name in the `Queue` objects, when we have a VM object itself.

When we have duplicate VM names on a system, the call to `mgmt.get_vm()` within `delete_vm()` fails with an exception. This addresses that by directly calling cleanup on the VM object that we already have, when we know its a duplicate as you have detected here.

If we step back and refactor to actually pass the VM objects between the scan and delete threads, then we eliminate the problem of trying to lookup the VM by its name.   This will also resolve problems in `scan_vm()`

Thoughts? I don't think it will be that large of a refactor, and it will allow for squashing some try/except blocks.

I would like to build on top of that by deleting VM's that we cannot get a creation date on,  as a default behavior.